### ISSUE TITLE
How child component can listen to parent event

### ISSUE THREAD

**Issue by ariaz** [pH200/cycle-react#38]
Question:

It is clear from the examples how to propagate an event from a child component to a parent (i.e todo-item -> todo-view from examples)

But I am having a hard time propagating an event from a parent to child. For example, consider dragging an element in a container. The mousedown is on the element, but I want the mousemove, mouseup from the parent. I understand this is doable by propagating the mousedown to the parent and adding the event on the parent, but I would like the event to live on the child. I.e,  onChildDrag makes more sense to live in child than on parent.

Thanks!

-Ariel

**Comment by pH200** [comment]
Hello Ariel. To simply answer your question: there's really not much difference between listening parent even from React components or from native DOM elements. Because the event is bubbling up by default, if you couldn't get the event from the child, you cannot get the event from the parent, too. The possible way to trigger/dispatch the event from the child is to listening the event from the parent, but dispatching the event by using child element. However, I wouldn't say that's listening the parent event from child element.

To further answer your question, if your component has some events which can be only triggered by its parent, maybe it's a better idea to move that event to the parent. For example, maybe onChildDrag can be the event from the container and send child item ID in its event parameter. 

Next, l think normally mousemove and mouseup can be listened in the child element for drag and drop. Maybe you can consider using [react-dragula](https://github.com/bevacqua/react-dragula) in that case. Cycle-react does support its ES2015 syntax.

Finally, for react and cycle-react components, there's only one way to send data into the component, that is by changing the props.
```
       ---------
      |         |
props > Element > events
      |         |
       ---------
```
So, basically it is possible to change the props from the parent, and then triggering the event from the child. Although I wouldn't recommend to do so.

**Comment by ariaz** [comment]
Thanks for the answer. I was hoping for a way to have globally accessible interactions as I don't want to force having all interaction logic living in the parent. For example, I would like to have a 'draggable' HOC so I wouldn't have to rewrite it every time. Consider the following HOC where
the mousedown is from HOC component, mousemove and mouseup are from its container

`const draggable = BaseComponent => 
  Cycle.component(BaseComponent.name, (interactions, props) => {

    const onObjectMouseDown = interactions.get('onObjectMouseDown')
    const onSceneMouseMove = interactions.get('onParentMouseMove')
    const onSceneMouseUp   = interactions.get('onParentMouseUp')

    const onObjectMove = onObjectMouseDown
      .switchMap( md => {
        const start = toPoint(md)
        return onSceneMouseMove
          .map( mm => ({mm, start}))
          .takeUntil(onSceneMouseUp)
      })
    const view = props.distinctUntilChanged().map( props => {
      return (
        <g
          onMouseDown={interactions.listener('onObjectMouseDown')}
        >
          <BaseComponent {...props}/>
        </g>
      )
    })
    const events = { onObjectMove }
  }
  return { view, events }
})`

**Comment by pH200** [comment]
Now the best way is to use Context from React. Please check [TodoMVC](https://github.com/pH200/cycle-react/tree/master/examples/web/todomvc) for example.
